### ISSUE TITLE
SqlServerProtocolTcpIp: New resource proposal

### ISSUE THREAD

**Issue by johlju** [dsccommunity/SqlServerDsc#1378]
### Description
This resource should manage IP address groups.

IP Address groups are added depending on available network cards, see [Adding or Removing IP Addresses](https://docs.microsoft.com/en-us/sql/tools/configuration-manager/tcp-ip-properties-ip-addresses-tab#adding-or-removing-ip-addresses). In my lab server I only have IP1-IP6 and IPAll. But in a test server with a cluster I have IP1-IP8 and IPAll.
So we cannot support adding and removing IP address groups, but I think we should support managing IP address groups with this resource. For that reason I see a need 
for a `IpAddressGroup` parameter that is mentioned in the https://github.com/PowerShell/SqlServerDsc/issues/339#issuecomment-298329440. The user can then choose which group should have which IP address, and if it should be enabled or not. This resource will most likely be used to change the IP addresses to different once that wasn't used during installation, or configure additional ones when network configuration changes.

It would be good to be able to support all these properties. The property `Active` seems to be read-only.
![image](https://user-images.githubusercontent.com/7189721/36933228-b69ad8f2-1ed5-11e8-8244-c0714dc939e0.png)

### Proposed properties
```
[ClassVersion("1.0.0.0"), FriendlyName("SqlServerProtocolTcpIp")]
class MSFT_SqlServerProtocolTcpIp : OMI_BaseResource
{
    [Write, Description("The host name of the SQL Server to be configured. Default value is $env:COMPUTERNAME.")] String ServerName;
    [Key, Description("The name of the SQLf instance to be configured.")] String InstanceName;
    [Key, Description("The name of the IP address group in the TCP/IP protocol.")] String IpAddressGroup;
    [Write, Description("Enables or disables the IP address. Only used if the IP address group is not set to 'IPAll'. If not specified, the existing value will not be changed.")] Boolean Enabled;
    [Write, Description("The IP address. Only used if the IP address group is not set to 'IPAll'. If not specified, the existing value will not be changed.")] String IPAddress;
    [Write, Description("Specifies whether the SQL Server instance should use a dynamic port. Value will be ignored if TcpPort is set to a non-empty string. If not specified, the existing value will not be changed.")] Boolean TcpDynamicPort;
    [Write, Description("The TCP port(s) that SQL Server should be listening on. If the IP address should listen on more than one port, list all ports separated with a comma ('1433,1500,1501'). If not specified, the existing value will not be changed.")] String TcpPort;
    [Write, Description("If set to $true then SQL Server and dependent services will be restarted if a change to the configuration is made. The default value is $false.")] Boolean RestartService;
    [Write, Description("Timeout value for restarting the SQL Server services. The default value is 120 seconds.")] UInt16 RestartTimeout;
}
```

### Special considerations or limitations

This is a redesign of the resource `SqlServerNetwork` and splitting it up into two new resources will simplify the code and therefore easier to maintain. This resource could be added side-by-side with `SqlServerNetwork` and together with `SqlServerProtocol` will deprecate the resource `SqlServerNetwork`.
This was discussed in closed PR https://github.com/PowerShell/SqlServerDsc/pull/1045.

**Comment by johlju** [comment]
There might be some code that can be reused in the closed PR https://github.com/PowerShell/SqlServerDsc/pull/1045.

**Comment by johlju** [comment]
The are some code in https://github.com/dsccommunity/SqlServerDsc/issues/339#issuecomment-298329440 that can be used to get test data. Also this issue also closes issue #339.
